### ISSUE TITLE
Enrich clinical reports with stop reason categories during runtime

### ISSUE THREAD

**Issue by ireneisdoomed** [opentargets/issues#4300]
As a **developer** I want **to apply the `opentargets/clinical_trial_reason_to_stop` model during the generation of the clinical reports dataset in PTS** because **this annotation enriches the original clinical report dataset**

## Background

We used to get predictions for the reasons to stop every time ChEMBL submitted a new evidence file.
Now that we are integrating `clinical_mining`, this annotation comes directly from clinical trials.

## Tasks

- [x] Load https://huggingface.co/opentargets/clinical_trial_stop_reasons and use classifier for all stop reasons
- [x] Convert logits to probabilities and apply a **threshold of 0.3** 
- [x] Mount predictions on to the clinical report with a new column `trialStopReasonCategories`

**Comment by ireneisdoomed** [comment]
## Results

Inference after running predictions in batches and caching the model loading takes 1min (yay!) in my local machine (accelerated with MPS)

- Clinical trials as of today: 149,804
- Clinical trials that stopped early: 14,813
- Clinical trials that stopped early with null categories: 0
- Clinical trials that stopped early and category is unassigned (`Uncategorised`): 317
- Distribution of categories:
```
[{'trialStopReasonCategories': ['Business_Administrative'], 'count': 4258},
 {'trialStopReasonCategories': ['Insufficient_Enrollment'], 'count': 3965},
 {'trialStopReasonCategories': ['Negative'], 'count': 1217},
 {'trialStopReasonCategories': ['Study_Design'], 'count': 645},
 {'trialStopReasonCategories': ['Invalid_Reason'], 'count': 597},
 {'trialStopReasonCategories': ['Logistics_Resources'], 'count': 556},
 {'trialStopReasonCategories': ['Covid19'], 'count': 497},
 {'trialStopReasonCategories': ['Safety_Sideeffects'], 'count': 453},
 {'trialStopReasonCategories': ['Study_Staff_Moved'], 'count': 430},
 {'trialStopReasonCategories': ['Another_Study'], 'count': 413},
 {'trialStopReasonCategories': ['Uncategorised'], 'count': 317},
 {'trialStopReasonCategories': ['Regulatory'], 'count': 277},
 {'trialStopReasonCategories': ['Business_Administrative',
   'Logistics_Resources'],
  'count': 153},
 {'trialStopReasonCategories': ['Logistics_Resources', 'Study_Design'],
  'count': 110},
 {'trialStopReasonCategories': ['Insufficient_Enrollment', 'Study_Design'],
  'count': 83},
 {'trialStopReasonCategories': ['Invalid_Reason', 'No_Context'], 'count': 77},
 {'trialStopReasonCategories': ['Business_Administrative',
   'Insufficient_Enrollment'],
  'count': 73},
 {'trialStopReasonCategories': ['Another_Study', 'Negative'], 'count': 60},
 {'trialStopReasonCategories': ['Business_Administrative', 'Invalid_Reason'],
  'count': 59},
 {'trialStopReasonCategories': ['Negative', 'Safety_Sideeffects'],
  'count': 57},
 {'trialStopReasonCategories': ['Another_Study', 'Study_Design'], 'count': 57},
 {'trialStopReasonCategories': ['Insufficient_Enrollment',
   'Logistics_Resources'],
  'count': 54},
 {'trialStopReasonCategories': ['No_Context'], 'count': 51},
 {'trialStopReasonCategories': ['Business_Administrative',
   'Insufficient_Enrollment',
   'Logistics_Resources'],
  'count': 41},
 {'trialStopReasonCategories': ['Insufficient_Enrollment', 'Invalid_Reason'],
  'count': 37},
 {'trialStopReasonCategories': ['Business_Administrative', 'Regulatory'],
  'count': 37},
 {'trialStopReasonCategories': ['Insufficient_Enrollment', 'Negative'],
  'count': 35},
 {'trialStopReasonCategories': ['Regulatory', 'Study_Design'], 'count': 34},
 {'trialStopReasonCategories': ['Business_Administrative',
   'Study_Staff_Moved'],
  'count': 24},
 {'trialStopReasonCategories': ['Another_Study', 'Business_Administrative'],
  'count': 16},
 {'trialStopReasonCategories': ['Insufficient_Enrollment',
   'Logistics_Resources',
   'Study_Design'],
  'count': 15},
 {'trialStopReasonCategories': ['Insufficient_Enrollment',
   'Study_Staff_Moved'],
  'count': 13},
 {'trialStopReasonCategories': ['Another_Study', 'Insufficient_Enrollment'],
  'count': 13},
 {'trialStopReasonCategories': ['Regulatory', 'Safety_Sideeffects'],
  'count': 12},
 {'trialStopReasonCategories': ['Business_Administrative', 'Study_Design'],
  'count': 9},
 {'trialStopReasonCategories': ['Logistics_Resources', 'Study_Staff_Moved'],
  'count': 8},
 {'trialStopReasonCategories': ['Negative', 'Study_Design'], 'count': 8},
 {'trialStopReasonCategories': ['Invalid_Reason', 'Study_Staff_Moved'],
  'count': 8},
 {'trialStopReasonCategories': ['Safety_Sideeffects', 'Study_Design'],
  'count': 8},
 {'trialStopReasonCategories': ['Business_Administrative', 'Negative'],
  'count': 6},
 {'trialStopReasonCategories': ['Another_Study',
   'Insufficient_Enrollment',
   'Study_Design'],
  'count': 5},
 {'trialStopReasonCategories': ['Invalid_Reason', 'Logistics_Resources'],
  'count': 4},
 {'trialStopReasonCategories': ['Business_Administrative',
   'Logistics_Resources',
   'Study_Design'],
  'count': 2},
 {'trialStopReasonCategories': ['No_Context', 'Regulatory'], 'count': 2},
 {'trialStopReasonCategories': ['Covid19', 'Logistics_Resources'], 'count': 2},
 {'trialStopReasonCategories': ['Business_Administrative',
   'Safety_Sideeffects'],
  'count': 2},
 {'trialStopReasonCategories': ['Another_Study', 'Safety_Sideeffects'],
  'count': 2},
 {'trialStopReasonCategories': ['Insufficient_Enrollment',
   'Negative',
   'Study_Design'],
  'count': 2},
 {'trialStopReasonCategories': ['Business_Administrative',
   'Logistics_Resources',
   'Study_Staff_Moved'],
  'count': 2},
 {'trialStopReasonCategories': ['Business_Administrative',
   'Invalid_Reason',
   'Regulatory'],
  'count': 1},
 {'trialStopReasonCategories': ['Insufficient_Enrollment',
   'Safety_Sideeffects'],
  'count': 1},
 {'trialStopReasonCategories': ['Insufficient_Enrollment',
   'Invalid_Reason',
   'Study_Design'],
  'count': 1},
 {'trialStopReasonCategories': ['Invalid_Reason', 'Study_Design'], 'count': 1},
 {'trialStopReasonCategories': ['Business_Administrative',
   'Invalid_Reason',
   'Logistics_Resources'],
  'count': 1},
 {'trialStopReasonCategories': ['Another_Study',
   'Insufficient_Enrollment',
   'Negative'],
  'count': 1},
 {'trialStopReasonCategories': ['Another_Study',
   'Negative',
   'Safety_Sideeffects'],
  'count': 1}]
```
